### PR TITLE
Add Document update and create APIs

### PR DESCRIPTION
**Pull request by micha91**
With this PR we add the high level APIs for document creation and updating documents. In addition, we introduce the Rendering Layouts to the Document class to enable control over WorkItem representations.

### CODE REVIEW

**Review by ewuerger — CHANGES_REQUESTED** [review]
Only one to resolve (`module_folder` for every document in the list). The others are pain points which could be addressed sometime else.

Then: LGTM.

File: polarion_rest_api_client/clients/documents.py [hunk 1]
```diff
@@ -64,27 +73,165 @@ def get(
...
     def get_multi(
         self, *args, page_size=100, page_number=1, **kwargs
     ) -> tuple[list[dm.Document], bool]:
         """Return a list of documents - Not implemented yet."""
         raise NotImplementedError
 
-    def _create(self, items: dm.Document | list[dm.Document]):
-        raise NotImplementedError
+    def _create(self, items: list[dm.Document]):
+        assert items[0].module_folder is not None, "module folder must be set"
```
[the remark this answers is not in the document]
Comment (micha91) [reply polarion_rest_api_client/clients/documents.py:177]: Nope, `_create` is called by from the `create` method of the super class. Before `_create` is called, all documents are grouped by their `module_folder` using the `_split_into_batches` method (see line 105+). For that reason all `module_folders` of `items` should have the same folder

File: tests/conftest.py [hunk 2]
```diff
@@ -24,6 +24,20 @@ def fixture_client():
     )
 
 
+@pytest.fixture(name="new_client")
+def fixture_new_client():
```
[the remark this answers is not in the document]
Comment (micha91) [reply tests/conftest.py:28]: see #41

File: polarion_rest_api_client/clients/documents.py [hunk 3]
```diff
@@ -64,27 +73,165 @@ def get(
...
+                    ),
+                ),
+            )
+        )
+
+        res = patch_document.sync_detailed(
+            project_id=self._project_id,
+            space_id=to_update.module_folder,
+            document_name=to_update.module_name,
+            client=self._client.client,
```
[the remark this answers is not in the document]
Comment (micha91) [reply polarion_rest_api_client/clients/documents.py:164]: see #42

File: polarion_rest_api_client/clients/documents.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (micha91) [reply polarion_rest_api_client/clients/documents.py:178]: a comment was added

**Review by ewuerger — APPROVED** [review]
LGTM

**Pull request merged**